go/vt/sqlparser/sql.y

```diff
@@ -7096,98 +7068,89 @@ function_call_keyword:
   }
-// these functions can also be called with an optional argument
-| CURRENT_TIMESTAMP func_datetime_precision
+| UTC_TIMESTAMP func_datetime_prec_opt
   {
-    $$ = &CurTimeFuncExpr{Name:NewColIdent(string($1)), Fsp:$2}
+    $$ = &FuncExpr{Name: NewColIdent(string($1)), Exprs: SelectExprs{&AliasedExpr{Expr: $2}}}
   }
-| UTC_TIMESTAMP func_datetime_precision
+| TIMESTAMPADD openb time_unit ',' value_expression ',' value_expression closeb
   {
-    $$ = &CurTimeFuncExpr{Name:NewColIdent(string($1)), Fsp:$2}
+    $$ = &TimestampFuncExpr{Name:string("timestampadd"), Unit:string($3), Expr1:$5, Expr2:$7}
   }
-| UTC_TIME func_datetime_precision
+| TIMESTAMPDIFF openb time_unit ',' value_expression ',' value_expression closeb
   {
-    $$ = &CurTimeFuncExpr{Name:NewColIdent(string($1)), Fsp:$2}
+    $$ = &TimestampFuncExpr{Name:string("timestampdiff"), Unit:string($3), Expr1:$5, Expr2:$7}
   }
-  // now
-| LOCALTIME func_datetime_precision
+| EXTRACT openb time_unit FROM value_expression closeb
   {
-    $$ = &CurTimeFuncExpr{Name:NewColIdent(string($1)), Fsp:$2}
+    $$ = &ExtractFuncExpr{Name: string($1), Unit: string($3), Expr: $5}
   }
-  // now
-| LOCALTIMESTAMP func_datetime_precision
+
+// functions that can be used with the ON UPDATE clause
+function_call_on_update:
+  // NOW is special; it can't be called without parentheses
+  NOW openb closeb
   {
-    $$ = &CurTimeFuncExpr{Name:NewColIdent(string($1)), Fsp:$2}
+    $$ = &FuncExpr{Name: NewColIdent(string($1))}
   }
-  // curtime
-| CURRENT_TIME func_datetime_precision
+| NOW openb INTEGRAL closeb
   {
-    $$ = &CurTimeFuncExpr{Name:NewColIdent(string($1)), Fsp:$2}
+    $$ = &FuncExpr{Name: NewColIdent(string($1)), Exprs: SelectExprs{&AliasedExpr{Expr: NewIntVal($3)}}}
   }
-| TIMESTAMPADD openb time_unit ',' value_expression ',' value_expression closeb
+| CURRENT_TIMESTAMP func_datetime_prec_opt
   {
-    $$ = &TimestampFuncExpr{Name:string("timestampadd"), Unit:string($3), Expr1:$5, Expr2:$7}
+    $$ = &FuncExpr{Name: NewColIdent(string($1)), Exprs: SelectExprs{&AliasedExpr{Expr: $2}}}
   }
-| TIMESTAMPDIFF openb time_unit ',' value_expression ',' value_expression closeb
+| LOCALTIME func_datetime_prec_opt
   {
-    $$ = &TimestampFuncExpr{Name:string("timestampdiff"), Unit:string($3), Expr1:$5, Expr2:$7}
+    $$ = &FuncExpr{Name: NewColIdent(string($1)), Exprs: SelectExprs{&AliasedExpr{Expr: $2}}}
   }
-| EXTRACT openb time_unit FROM value_expression closeb
+| LOCALTIMESTAMP func_datetime_prec_opt
   {
-    $$ = &ExtractFuncExpr{Name: string($1), Unit: string($3), Expr: $5}
+    $$ = &FuncExpr{Name: NewColIdent(string($1)), Exprs: SelectExprs{&AliasedExpr{Expr: $2}}}
   }
 
+
 // Optional parens for certain keyword functions that don't require them.
 func_parens_opt:
   /* empty */
 | openb closeb
 
-func_datetime_precision:
-  openb value_expression closeb
+// Optional datetime precision for certain functions.
+func_datetime_prec_opt:
```

Comment:
do we have error tests with non-integer precisions?

Reply:
Yes, they are here:
https://github.com/dolthub/vitess/pull/296/files#diff-f888604778470e9f0188a9034ce01275a29c0bc35a2edc76b3b08ae279ef9f69R4518-R4549